oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/NimbusJwtEncoder.java

```diff
@@ -124,6 +131,10 @@ private JWK selectJwk(JwsHeader headers) {
 					"Failed to select a JWK signing key -> " + ex.getMessage()), ex);
 		}
 
+		if (null != this.jwkSelector) {
```

Comment:
Instead of checking for null, set `this.jwkSelector` to have a default and then do:

```java
return this.jwkSelector.convert(jwks);
```

You can then remove the rest of this method.